test/testing/test_services.py

```diff
@@ -31,6 +35,23 @@ def reset_mock():
     handle_event.reset_mock()
 
 
+@pytest.yield_fixture
+def spawn_thread():
+
+    threads = []
+
+    def spawn(fn, *args):
+        threads.append(eventlet.spawn(fn, *args))
+
+    yield spawn
+
+    for gt in threads:
+        try:
+            gt.kill()
+        except:
```

Comment:
`GreenletExit`? That inherits from `BaseException`, but it's not raised from `gt.kill()` (only `gt.wait()`).

I think `except Exception` is what we want here